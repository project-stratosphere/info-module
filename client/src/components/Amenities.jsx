import PropTypes from 'prop-types';
import React from 'react';
import { SVGContainer } from './styles/Summary.styles';
import {
  BasicSVG,
  DiningSVG,
  FacilitiesSVG,
  GuestSVG,
  BnbSVG,
  SafetySVG,
} from './SVG';
import {
  Wrapper,
  Title,
  AmenitiesContainer,
  CategoryContainer,
  CategoryHead,
  CategoryItemContainer,
  AmenityDescription,
  Supplemental,
  Modal,
  PreviewContainer,
  ShowAll,
  PreviewAmenity,
  PreviewTitle,
  ModalContent,
} from './styles/Amenities.styles';

const renderIcon = (category) => {
  if (category === 'Dining') {
    return (
      <SVGContainer>
        <DiningSVG />
      </SVGContainer>
    );
  } else if (category === 'Facilities') {
    return (
      <SVGContainer>
        <FacilitiesSVG />
      </SVGContainer>
    );
  } else if (category === 'Guest access') {
    return (
      <SVGContainer>
        <GuestSVG />
      </SVGContainer>
    );
  } else if (category === 'Bed and bath') {
    return (
      <SVGContainer>
        <BnbSVG />
      </SVGContainer>
    );
  } else if (category === 'Safety features') {
    return (
      <SVGContainer>
        <SafetySVG />
      </SVGContainer>
    );
  }
  return (
    <SVGContainer>
      <BasicSVG />
    </SVGContainer>
  );
};

const amenitiesCount = (amenities) => {
  console.log(amenities)
  let count = 0;
  for (let i = 0; i < amenities.length; i += 1) {
    count += amenities[i].category_items.length;
  }
  return count;
};

export default class Amenities extends React.Component {
  constructor(props) {
    super(props);
    this.handleAmenitiesDisplay = this.handleAmenitiesDisplay.bind(this);
    this.handleAmenitiesHide = this.handleAmenitiesHide.bind(this);
    this.state = {
      displayModal: 'none',
    };
  }

  handleAmenitiesDisplay() {
    this.setState({
      displayModal: 'flex',
    });
  }

  handleAmenitiesHide() {
    this.setState({
      displayModal: 'none',
    });
  }

  render() {
    return (
      <Wrapper>
        <PreviewContainer>
          <PreviewTitle>
            Amenitites
          </PreviewTitle>
          {
            this.props.amenities.items.map(item => (
              <PreviewAmenity
                key={item.category_head}
              >
                {renderIcon(item.category_head)}
                {item.category_items[0].amenity_description}
              </PreviewAmenity>
            ))
          }
          <ShowAll
            onClick={this.handleAmenitiesDisplay}
            className="modal-display"
          >
            Show all {amenitiesCount(this.props.amenities.items)} amenities
          </ShowAll>
        </PreviewContainer>
        <Modal
          displayModal={this.state.displayModal}
          onClick={this.handleAmenitiesHide}
          className="modal-hide"
        >
          <ModalContent>
            <Title>
              Amenities
            </Title>
            <AmenitiesContainer>
              {
                this.props.amenities.items.map(item => (
                  <CategoryContainer
                    key={item.category_head}
                  >
                    <CategoryHead>
                      {item.category_head}
                    </CategoryHead>
                    {
                      item.category_items.map(categoryItem => (
                        <CategoryItemContainer
                          key={categoryItem.amenity_description}
                        >
                          <AmenityDescription>
                            {categoryItem.amenity_description}
                          </AmenityDescription>
                          <Supplemental>
                            {categoryItem.supplemental_description}
                          </Supplemental>
                        </CategoryItemContainer>
                      ))
                    }
                  </CategoryContainer>
                ))
              }
            </AmenitiesContainer>
          </ModalContent>
        </Modal>
      </Wrapper>);
  }
}

Amenities.propTypes = {
  amenities: PropTypes.shape({
    id: PropTypes.number,
    items: PropTypes.arrayOf(PropTypes.shape({
      category_head: PropTypes.string,
      category_items: PropTypes.arrayOf(PropTypes.shape({
        amenity_description: PropTypes.string,
        supplemental_description: PropTypes.string,
      })),
    })),
  }),
};

Amenities.defaultProps = {
  amenities: {
    id: 1,
    items: [
      {
        category_head: 'String',
        category_items: [
          {
            amenity_description: 'String',
            supplemental_description: 'String',
          },
        ],
      },
    ],
  },
};
