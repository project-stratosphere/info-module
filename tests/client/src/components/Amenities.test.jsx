import React from 'react';
import Enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Amenities from '../../../../client/src/components/Amenities';

Enzyme.configure({ adapter: new Adapter() });

const amenities = {
  id: 1,
  items: [
    {
      category_head: 'Mock Category Head',
      category_items: [
        {
          amenity_description: 'Mock Category Item',
          supplemental_description: 'Mock Supplemental Description',
        },
      ],
    },
  ],
};

describe('Amenities', () => {
  it('should render correctly', () => {
    const output = Enzyme.shallow(<Amenities
      amenities={amenities}
    />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should change state on modal display and modal hide', () => {
    const output = Enzyme.shallow(<Amenities
      amenities={amenities}
    />);
    expect(output.state().displayModal).toEqual('none');
    output.find('.modal-display').simulate('click');
    expect(output.state().displayModal).toEqual('flex');
    output.find('.modal-hide').simulate('click');
    expect(output.state().displayModal).toEqual('none');
  });
});
