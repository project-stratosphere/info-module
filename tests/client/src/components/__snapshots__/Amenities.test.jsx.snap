// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amenities should render correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Amenitites
    </styled.div>
    <styled.div
      key="Mock Category Head"
    >
      <styled.div>
        <BasicSVG />
      </styled.div>
      Mock Category Item
    </styled.div>
    <styled.div
      className="modal-display"
      onClick={[Function]}
    >
      Show all 
      1
       amenities
    </styled.div>
  </styled.div>
  <styled.div
    className="modal-hide"
    displayModal="none"
    onClick={[Function]}
  >
    <styled.div>
      <styled.div>
        Amenities
      </styled.div>
      <styled.div>
        <styled.div
          key="Mock Category Head"
        >
          <styled.div>
            Mock Category Head
          </styled.div>
          <styled.div
            key="Mock Category Item"
          >
            <styled.div>
              Mock Category Item
            </styled.div>
            <styled.div>
              Mock Supplemental Description
            </styled.div>
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
